crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -4804,72 +4788,70 @@ impl<'a> Checker<'a> {
                     }
                     _ => {}
                 }
-
-                return;
             }
-
-            first_iter = false;
-            import_starred = import_starred || scope.uses_star_imports();
-        }
-
-        if import_starred {
-            // F405
-            if self
-                .settings
-                .rules
-                .enabled(Rule::UndefinedLocalWithImportStarUsage)
-            {
-                let sources: Vec<String> = self
-                    .ctx
-                    .scopes
-                    .iter()
-                    .flat_map(Scope::star_imports)
-                    .map(|StarImportation { level, module }| {
-                        helpers::format_import_from(*level, *module)
-                    })
-                    .sorted()
-                    .dedup()
-                    .collect();
-                self.diagnostics.push(Diagnostic::new(
-                    pyflakes::rules::UndefinedLocalWithImportStarUsage {
-                        name: id.to_string(),
-                        sources,
-                    },
-                    expr.range(),
-                ));
+            ResolvedReference::ImplicitGlobal => {
+                // Nothing to do.
             }
-            return;
-        }
-
-        if self.settings.rules.enabled(Rule::UndefinedName) {
-            // Allow __path__.
-            if self.path.ends_with("__init__.py") && id == "__path__" {
-                return;
+            ResolvedReference::StarImport => {
+                // F405
+                if self
+                    .settings
+                    .rules
+                    .enabled(Rule::UndefinedLocalWithImportStarUsage)
+                {
+                    let sources: Vec<String> = self
+                        .ctx
+                        .scopes
+                        .iter()
+                        .flat_map(Scope::star_imports)
+                        .map(|StarImportation { level, module }| {
+                            helpers::format_import_from(*level, *module)
+                        })
+                        .sorted()
+                        .dedup()
+                        .collect();
+                    self.diagnostics.push(Diagnostic::new(
+                        pyflakes::rules::UndefinedLocalWithImportStarUsage {
+                            name: id.to_string(),
+                            sources,
+                        },
+                        expr.range(),
+                    ));
+                }
             }
+            ResolvedReference::NotFound => {
+                // F821
+                if self.settings.rules.enabled(Rule::UndefinedName) {
+                    // Allow __path__.
+                    if self.path.ends_with("__init__.py") && id == "__path__" {
```

Comment:
I realize that this code is unchanged in the PR, but why is `__path__` gated to `__init__.py` files?

Reply:
I think it's only set for modules that are packages?

Reply:
sorry i mixed this up with `__file__`